src/xtz_validator.ts

```diff
@@ -0,0 +1,27 @@
+import { base58_to_binary } from "base58-js";
+
+import { TDecode, TDecodeValidator } from "./types/validators.types";
+
+const getDecoded: TDecode = (address) => {
+  try {
+    return base58_to_binary(address);
+  } catch (e) {
+    // if decoding fails, assume invalid address
+    return null;
+  }
+};
+
+const xtzValidator: TDecodeValidator = {
+  isValidAddress: (address) => {
+    const decoded = getDecoded(address);
+
+    if (!decoded || !ArrayBuffer.isView(decoded) || decoded.length !== 27) {
```

Comment:
Should we declare something like this - `XTZ_DECODED_LENGTH`

```suggestion
    if (!decoded || !ArrayBuffer.isView(decoded) || decoded.length !== XTZ_DECODED_LENGTH) {
```

Reply:
Good question, it could be a good idea, but it would require to apply the same change for rest fo validators and it could be hacky sometimes as there are weird "calculations" or conditions like substrining before checking length etc so imo I would leave this without declaring as separated consts as it would be local only.